lib/ddtrace/opentracer/tracer.rb

```diff
@@ -56,6 +56,28 @@ def start_active_span(operation_name,
                             tags: nil,
                             ignore_active_scope: false,
                             finish_on_close: true)
+
+        # When meant to automatically determine the parent,
+        # Use the active scope first, otherwise fall back to any
+        # context generated by Datadog, so as to append to it and gain
+        # the benefit of any out-of-the-box tracing from Datadog preceding
+        # the OpenTracer::Tracer.
+        #
+        # We do this here instead of in #start_span because #start_span generates
+        # spans that are not assigned to a scope, a.k.a not supposed to be used by
+        # subsequent spans implicitly. By using the existing Datadog context, the span
+        # effectively ends up "assigned to a scope", by virtue of being added to the
+        # Context. Hence, it would behave more like an active span, which is why it
+        # should only be here.
+        unless child_of || ignore_active_scope
```

Comment:
This block is responsible for allowing OpenTracer spans to append to traces that were created by Datadog auto-instrumentation. e.g. `datadog.span` --> `open_tracer.span`.